### PR TITLE
feat(web-app): glossary component

### LINKED ISSUE #612: GlossaryComponent  MDX component
https://carbondesignsystem.com/guidelines/content/action-labels

`<GlossaryComponent />`

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #612 

#### Testing / reviewing

http://localhost:3000/mdx/glossary
https://carbondesignsystem.com/guidelines/content/action-labels

### CODE REVIEW

**Review by francinelucca — COMMENTED** [review]
-

File: services/web-app/components/glossary/glossary-list/glossary-list.module.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
+@use '@carbon/react/scss/spacing' as spacing;
+@use '@carbon/react/scss/type' as type;
+@use '@carbon/react/scss/theme' as theme;
+@use '@carbon/react/scss/utilities/convert' as convert;
+@use '@carbon/react/scss/breakpoint' as breakpoint;
+
+.glossary-entry {
+  padding-top: convert.rem(36px);
```
Comment (alisonjoseph) [inline services/web-app/components/glossary/glossary-list/glossary-list.module.scss:8]:
```suggestion
  padding-top: convert.rem(36px);
  color: theme.$text-primary;
```

minor, if we add this is works for all themes.

File: services/web-app/components/glossary/glossary.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <div className="glossary">
+      <GlossaryNav />
+      <div className="cds--row">
+        <div className="cds--col-lg-7">
+          <GlossaryList glossary={glossary} />
+        </div>
+        <div className="cds--col-lg-1" />
+      </div>
+    </div>
```
Comment (alisonjoseph) [inline services/web-app/components/glossary/glossary.js:21]:
```suggestion
    <Grid>
      <Column sm={4} md={8} lg={7}>
        <div className="glossary">
          <GlossaryNav />
          <GlossaryList glossary={glossary} />
        </div>
      </Column>
    </Grid>
```

Let's replace with updated grid components.

File: services/web-app/components/glossary/glossary.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
```
Comment (alisonjoseph) [inline services/web-app/components/glossary/glossary.js:6]:
```suggestion
 */
 import { Column, Grid } from '@carbon/react'
```

File: services/web-app/pages/mdx/index.mdx [hunk 4]
```diff
@@ -687,3 +687,11 @@ hac habitasse platea dictumst.
...
 
 <ColorPalette type="status-extended" shouldShowControls={false} />
+
+## Glossary
+
+<Grid>
+  <Column lg={7}>
+    <Glossary />
+  </Column>
+</Grid>
```
Comment (alisonjoseph) [inline services/web-app/pages/mdx/index.mdx:697]:
```suggestion
<Glossary />
```

File: services/web-app/pages/mdx/glossary.mdx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+## Glossary
+
+<Grid>
+  <Column lg={7}>
+    <Glossary />
+  </Column>
+</Grid>
```
Comment (alisonjoseph) [inline services/web-app/pages/mdx/glossary.mdx:7]:
```suggestion
<Glossary />
```

File: services/web-app/stories/components/glossary.stories.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+export default stories
+
+const Template = (args) => (
+  <Grid>
+    <Column lg={7}>
+      <Glossary {...args} />
+    </Column>
+  </Grid>
```
Comment (alisonjoseph) [inline services/web-app/stories/components/glossary.stories.js:23]:
```suggestion
  <Glossary />
```

File: services/web-app/components/glossary/glossary-nav/glossary-nav.module.scss [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    width: 100%;
+    padding: 0;
+    list-style-type: none;
+    transition: none;
+
+    a {
+      @include type.type-style('body-compact-02');
+
+      display: block;
+
```
Comment (alisonjoseph) [inline services/web-app/components/glossary/glossary-nav/glossary-nav.module.scss:42]:
```suggestion
      display: block;
```

File: services/web-app/components/glossary/glossary-list/glossary-list.module.scss [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+.page-h4 {
+  width: 100%;
+  margin-top: spacing.$spacing-04;
+  @include type.type-style('heading-02');
+  @include breakpoint.breakpoint('md') {
+    width: 75%;
+  }
+  @include breakpoint.breakpoint('lg') {
+    width: 66.667%;
+  }
```
Comment (alisonjoseph) [inline services/web-app/components/glossary/glossary-list/glossary-list.module.scss:47]:
```suggestion
```

we shouldn't need this, this is leftover overriding something from Gatsby I believe.

File: services/web-app/components/glossary/glossary-list/glossary-list.module.scss [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+  padding: 0;
+  margin-top: spacing.$spacing-02;
+
+  color: theme.$text-secondary;
+  font-style: italic;
+}
+
+.page-h4 {
+  width: 100%;
```
Comment (alisonjoseph) [inline services/web-app/components/glossary/glossary-list/glossary-list.module.scss:39]:
```suggestion
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Just a few styling cleanup things, this looks great!

File: services/web-app/components/glossary/glossary-nav/glossary-nav.module.scss (hunk 7 above)
Comment (francinelucca) [reply services/web-app/components/glossary/glossary-nav/glossary-nav.module.scss:42]: what's the diff here? 😅

Comment (alisonjoseph) [reply services/web-app/components/glossary/glossary-nav/glossary-nav.module.scss:42]: 😆 ignore this one haha

File: services/web-app/components/glossary/glossary.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+import { Column, Grid } from '@carbon/react'
+
+import GlossaryList from './glossary-list'
+import GlossaryNav from './glossary-nav'
+
+const Glossary = () => {
+  const glossary = require('@/data/glossary')
+  return (
+    <Grid>
+      <Column sm={4} md={8} lg={7}>
```
Comment (alisonjoseph) [inline services/web-app/components/glossary/glossary.js:16]:
```suggestion
      <Column sm={4} md={6} lg={7}>
```
to match the paragraph/component width at tablet.

**Review by alisonjoseph — DISMISSED** [review]
LGTM other than one small column width update at md breakpoint! 🥳

File: services/web-app/components/glossary/glossary.js (hunk 10 above)
Comment (francinelucca) [reply services/web-app/components/glossary/glossary.js]: https://carbondesignsystem.com/guidelines/content/action-labels uses 8 columns at md, is it ok if we change that?
image

Comment (alisonjoseph) [reply services/web-app/components/glossary/glossary.js]: Yea, I'm confident that's a mistake / oversight on the current website.

**Review by alisonjoseph — APPROVED** [review]
seal-of-approval2

**Review by andreancardona — APPROVED** [review]
🔥

**Pull request merged**